### ISSUE TITLE
Item detail does not work with Doctrine entity

### ISSUE THREAD

**Issue by xpackal5** [contributte/datagrid#835]
Hi I am using grid with QueryBuilder datasource, but item detail does not work correctly.
I have 
```
        $grid->setItemsDetail();
        $grid->setTemplateFile(__DIR__ . "/templates/projectDetail.latte");
```
and template
```
{extends $originalTemplate}
{block detail}
<p>{$item->getComment()}</p>
{/block}
```
When I click on eye icon, detail is shown but all rows disseapear.

**Comment by juniwalk** [comment]
@xpackal5 Hello, is this still actual? I am actually using item details with doctrine without issues.

**Comment by darkWolf-PR** [comment]
Hi, @juniwalk 

I´m getting the same outcome using Nette Database Table Selection as datasource and showing just single column as detail like this:

`$grid->setItemsDetail(function($item) { return $item->content; });`

Eye button in actions shows normally, but after click, all table rows gets removed, only the detail is shown between header and footer of the datagrid. (using only base template from datagrid, no special formatting extended template)

It´s just some preformatted content in that column so I thought I´ll just show it this way instead of making another page with detail, but it seems like I´ll have to do that detail actually.

**Comment by juniwalk** [comment]
@darkWolf-PR have you checked any issues in JS console? XHR request? I just went into random datagrid I have, put your code there just calling `$item->getName();` and it worked immidietly.

I tried with my custom template and with original one too. Both work.

**Comment by darkWolf-PR** [comment]
Yep, checked, no JS issue, XHR request seems ok, at least as much as I can say, theres a lot of JSON stuff.

Address looks like this:
`http://web.local/admin/contacts-archive?contactsArchiveGrid-grid-id=3&contactsArchiveGrid-grid-perPage=20&contactsArchiveGrid-grid-sort%5Bentry_date%5D=DESC&do=contactsArchiveGrid-grid-getItemDetail`

I have 3 testing rows in table, after click, it changes them with detail block, instead of adding the detail block. So it could be connected with append/prepend. I´m using Naja-maybe something with Naja support? 
Also, I have admin completely ajaxified, all links ajaxed, but everything else(that I´ve tried till now) seems to be working with Datagrid, so that should not be a problem.

**Comment by juniwalk** [comment]
@darkWolf-PR Oh, I am still stuck on nette.ajax.js, I haven't tried Naja yet. Could you create some small sandbox where it does not work so I could play with it?

**Comment by darkWolf-PR** [comment]
@juniwalk I can try during weekend, not sure if I manage, I´ve never actually uploaded any project to github...

Maybe we can ping someone who did naja support for datagrid - not sure who did it. It kind of look like it could be the issue. 

Sadly @xpackal5 did not say if hes been using nette.ajax or naja.

**Comment by juniwalk** [comment]
@darkWolf-PR If you don't want to dive into uploading project into github (it is great opportunity though :)) you could send me the sandbox using any other way that is comfortable for you.

**Comment by darkWolf-PR** [comment]
Iˇve tried switching naja with nette.ajax just for a test. No difference. So it has to be connected with something else. I´ll try, if I manage to make some sandbox with same outcome.

**Comment by darkWolf-PR** [comment]
So, I´ve uploaded test version here:

[https://github.com/darkWolf-PR/DatagridDetailTest](url)

No modules and other stuff, but otherwise, pretty much same as what I have in my project. 1st I thought I will just use an array datasource, but It just doesn´t work...details were not showing at all. Even with just `$grid->setItemsDetail(function() { return 'Lorem Ipsum'; });`, datagrid shown "no items found". No idea why is that... Switching to nette database as datasource, it redraws all rows and shows just detail. I´ve set-up full content ajax with Naja the same way I have it set-up in my admin.

**Comment by juniwalk** [comment]
@darkWolf-PR Alright, i will try to work on in when I get some free time, what is your enviroment like? php version, db type, version etc.

**Comment by darkWolf-PR** [comment]
@juniwalk Im using wampserver with PHP 7.4.10, MariaDB 10.3, Apache 2.4.46

**Comment by juniwalk** [comment]
@darkWolf-PR So all I did to make it work is to remove the `{snippet content}`. I don't know why but it redraws the whole content instead of just the single row with the item detail.

I don't use snippet on the whole content that is probably why I don't get the issue with this.

**Comment by darkWolf-PR** [comment]
You mean in layout.latte? That´s what I have for ajaxified pages in admin. It´s basically from nette.ajax.history setup as I´ve been using that before switching to Naja.

Datagrid seems to have problems with thit overal...that´s why I´m already using this code:

```php
if (!isset($this->getPayload()->_datagrid_url)) {
                $this->redrawControl('title');
                $this->redrawControl('flashes');
                $this->redrawControl('content');
            }
```

Without that, datagrid was not working at all. But it doesn´t seem to catch stuff behind showing details. I´ve been messing with it already while testing this as I thought It could be a source of problem. But I had no success overcoming it.

**Comment by juniwalk** [comment]
@darkWolf-PR I see, like I said I don't use snippet over the whole content, I don't ajaxify whole page reloading.
